clang/lib/CodeGen/CGObjCMac.cpp

```diff
@@ -2220,6 +2220,14 @@ CGObjCCommonMac::EmitMessageSend(CodeGen::CodeGenFunction &CGF,
   RValue rvalue = CGF.EmitCall(MSI.CallInfo, Callee, Return, ActualArgs,
                                &CallSite);
 
+  // Set type identifier metadata of indirect calls for call graph section.
+  if (CGM.getCodeGenOpts().CallGraphSection && Method && CallSite &&
```

Comment:
I don't think CallSite can be null here

Reply:
Ack. This change should not have been made for ObjC. I will remove this snippet of code that makes the call to CreateFunctionTypeMetadataforIcall which should also address this comment. I will upload the change along with the refactoring change suggested by Eli as soon as I can. Thank you.